pytest_splunk_addon_ui_smartx/components/table.py

```diff
@@ -547,3 +560,22 @@ def check_alert_sign(self, row_name, column_name="account"):
             return True
         except exceptions.NoSuchElementException:
             return False
+
+    def __getattr__(self, key):
+        """
+        Makes the web-elements to be accessible directly.
+        - For example self.elements = {"textbox": Selector(by=..., select=...),
+            Access the element by doing self.textbox directly.
+        - It also has implicit wait while finding the element.
+            :param key: The key of the element mentioned in self.elements
+            :returns: The webelement we are accessing
+        """
+        # NOTE: Overriding the implementation for only table component.
+        try:
+            return self.get_element(key)
+        except KeyError:
+            raise
+        except exceptions.TimeoutException:
+            # in case when the element isn't found, return None
+            # so that checks based on this class's properties are in sync.
+            return None
```

Comment:
Can you please explain a bit more why do we need this method for this particular PR?

Reply:
This snippet is called when we do something like `self.<element_on_page>` which internally goes to find that element on the page. With this change, when the element/component isn't found:
- In expected scenarios (when the element should be absent), returning `None` for `self.<element_on_page>` execution, the SmartX framework has clean exit, the add-on's test case works as expected.
- In unexpected scenarios (when the element should be present), returning `None` would not return the required "string" values of the element and the add-on developers would have to update their test case or check their add-on's globalConfig.

Without this change, if we keep a subset of table actions (any combination of "edit, clone, delete" but not all) for input table or configuration table, the SmartX framework would break.